### DISCUSSION TITLE
How to add visual objects?

### DISCUSSION THREAD

**Discussion by waleedbkhalid** [raisimTech/raisimLib#569] [General]
I have declared a ball variable as:

raisim::Sphere*ball;

In the private section of my environment file.

Then in my constructor I do this:
ball = world_.addSphere(0.05, 2.0,"default" );
    
Then in my reset function I do this:

  ball->setPosition(raisim::Vec<3>{goal_X_coordinate,goal_Y_coordinate,0});
    ball->setLinearVelocity(raisim::Vec<3>{0,0,0});
    ball->setBodyType(raisim::BodyType::STATIC);

The issue is, I only need a visual not an actual sphere. 

So I went to the visual objects,

But I am not sure, how do I declare a visualSphere, then initialize it in constructor and then set its properties.

I have tried a few things and got type errors. Can you please provide guidance for this?

I dont want to do it using an auto pointer, as that pointer is not accessible in functions. 

Secondly,
There is a visualHeightMap available, which takes in x and y range and the height vector corresponding to that.

My question is, is there a method similar to setPosition that sets the x and y range and height range. This functionality would help me edit the heightMap at the right position in my code.

**Comment by jhwangbo** [comment]
You can check the return type of addVisualSphere

**Reply by waleedbkhalid** [reply]
So, I just fixed this. The return type of addVisualSphere is Visuals

and I did this

  raisim::Visuals*visualSphere;
  
  So it worked.
  
  But the return type of visualHeightMapData is
  
  HeightMapVisual 
  
  and that is not working.

**Reply by jhwangbo** [reply]
Error at the compile time? Whats the error message?

**Reply by waleedbkhalid** [reply]
raisim::HeightMapVisual*hmNavigation;

This is what i declared. 
/home/waleed/raisim_workspace/MPC_IL_RL/src/env/debug_app.cpp:6:
    /home/waleed/raisim_workspace/MPC_IL_RL/src/env/Environment.hpp:1218:11: error: ‘HeightMapVisual’ in namespace ‘raisim’ does not name a type
     1218 |   raisim::HeightMapVisual*hmNavigation;
          |           ^~~~~~~~~~~~~~~
    In file included from

**Reply by waleedbkhalid** [reply]
raisim::Visuals*visualSphere;

This is working and so are the associated methods. I meant the above line only. the visualHeightMap is not working and the error is shown above.
